packages/csv-parser-discount-code/src/main.js

```diff
@@ -0,0 +1,101 @@
+/* @flow */
+import type {
+  LoggerOptions,
+  ParseOptions,
+  ParserSummary,
+ } from 'types/discountCodes'
+import _ from 'lodash'
+import csv from 'csv-parser'
+import JSONStream from 'JSONStream'
+import highland from 'highland'
+import npmlog from 'npmlog'
+import { unflatten } from 'flat'
+
+export default class CsvParser {
+  // set flowtype annotations
+  delimiter: string;
+  multiValueDelimiter: string;
+  continueOnProblems: boolean;
+  logger: LoggerOptions;
+  _summary: ParserSummary;
+  _cartDiscountsToArray: Function;
+  _handleErrors: Function;
+  _rowIndex: number;
+
+
+  // should take in optional parameters: a logger and a configuration object
+  constructor (
+    logger: LoggerOptions,
+    options: ParseOptions = {},
+  ) {
+    const defaultOptions = {
+      delimiter: ',',
+      multiValueDelimiter: ';',
+      continueOnProblems: false,
+    }
+    Object.assign(this, defaultOptions, options)
+
+    this.logger = logger || {
+      error: npmlog.error.bind(this, ''),
+      warn: npmlog.warn.bind(this, ''),
+      info: npmlog.info.bind(this, ''),
+      verbose: npmlog.verbose.bind(this, ''),
+    }
+
+    this._cartDiscountsToArray = this._cartDiscountsToArray.bind(this)
+    this._handleErrors = this._handleErrors.bind(this)
+
+    this._rowIndex = 0
+  }
+
+  // Remove fields with empty values from the code objects
+  static _removeEmptyFields (item: Object) {
+    return _.omitBy(item, val => val === '')
+  }
+
+  parse (input: stream$Readable, output: stream$Writable) {
```

Comment:
Yeah... `ReadableStream` and `WritableStream` classes do not have the `.emit()` (according to flow), so this was the only way